eng/pipelines/templates/jobs/archetype-sdk-tests-jobs.yml

```diff
@@ -0,0 +1,117 @@
+parameters:
+- name: Platforms
+  type: object
+  default: []
+- name: PreSteps
+  type: object
+  default: []
+- name: PostSteps
+  type: object
+  default: []
+- name: EnvVars
+  type: object
+  default: {}
+- name: MaxParallel
+  type: number
+  default: 0
+- name: BuildInParallel
+  type: boolean
+  default: true
+- name: TimeoutInMinutes
+  type: number
+  default: 60
+- name: Location
+  type: string
+  default: ''
+- name: SubscriptionConfiguration
+  type: string
+  default: $(sub-config-azure-cloud-test-resources)
+- name: ServiceDirectory
+  type: string
+  default: not-specified
+
+jobs:
+  - ${{ each platform in parameters.Platforms }}:
+    - job:
+      displayName: ${{ platform.DisplayName }}
+      variables:
+        - template: ../variables/globals.yml
+        - name: SubscriptionConfiguration
+          value: ${{ parameters.SubscriptionConfiguration }}
+        - name: Location
+          value: ${{ parameters.Location }}
+        - name: OSVmImage
+          value: ${{ platform.OSVmImage }}
+        - name: TestTargetFramework
+          value: ${{ platform.TestTargetFramework }}
+
+      timeoutInMinutes: ${{ parameters.TimeoutInMinutes }}
+
+      pool:
+        vmImage: $(OSVmImage)
+      ${{ if platform.Container }}:
+        container: ${{ platform.Container }}
+
+      steps:
+        - ${{ if platform.PreSteps }}:
+          - ${{ platform.PreSteps }}
```

Comment:
I like this. We can have both platform specific and global pre-steps. 

I think the sequence is correct for the current use case as well as most other potential use cases. That is, platform-specific preparation may be required before running common pre-steps but it's less likely to have to run global pre-steps before some platform-specific step.